### PR TITLE
Make SplitTextCarrier work with superset maps

### PR DESCRIPTION
**Pull request by bensigelman**
For example, passing Go's http.Header map into both TracerState and Baggage should work. Related to https://github.com/opentracing/opentracing.github.io/issues/73.

### CODE REVIEW

File: propagation_ot.go [hunk 1]
```diff
@@ -53,7 +58,8 @@ func (p *splitTextPropagator) Inject(
 		splitTextCarrier.Baggage = make(map[string]string, l)
 	}
 	for k, v := range sc.raw.Baggage {
-		splitTextCarrier.Baggage[k] = v
+		fmt.Println("BHS2", prefixBaggage+k, v)
```
Comment (yurishkuro) [inline propagation_ot.go]: testing leftover?

**Comment by yurishkuro** [comment]
lgtm, but the tests are failing in the daperish example

File: propagation_ot.go (hunk 1 above)
Comment (bensigelman) [inline propagation_ot.go]: ah indeed -- thanks.

Will look at the dapperish test -- I was running `make test` and assuming that was all that needed to be done.

```
$ make test
go test -v -cover ./...
=== RUN   TestSpanPropagator
--- PASS: TestSpanPropagator (0.00s)
PASS
coverage: 67.8% of statements
ok  	github.com/opentracing/basictracer-go	0.018s	coverage: 67.8% of statements
?   	github.com/opentracing/basictracer-go/examples	[no test files]
?   	github.com/opentracing/basictracer-go/examples/dapperish	[no test files]
=== RUN   TestInMemoryRecorderSpans
--- PASS: TestInMemoryRecorderSpans (0.00s)
PASS
coverage: 100.0% of statements
ok  	github.com/opentracing/basictracer-go/testutils	0.017s	coverage: 100.0% of statements
```

Comment (bensigelman) [inline propagation_ot.go]: ah -- it's the change to arg counts for the context stuff.

**Pull request merged**